src/Stratis.Sidechains.Networks/CirrusMain.cs

```diff
@@ -39,7 +39,7 @@ internal CirrusMain()
             this.DefaultConfigFilename = NetworkDefaultConfigFilename;
             this.MaxTimeOffsetSeconds = 25 * 60;
 
-            var consensusFactory = new SmartContractPoAConsensusFactory();
+            var consensusFactory = new SmartContractCollateralPoAConsensusFactory();
```

Comment:
So all nodes (not only block producers) check collateral?

Reply:
No, only block producers